tpmeventlog/extract.go

```diff
@@ -12,69 +12,59 @@
 // License for the specific language governing permissions and limitations under
 // the License.
 
-// Package tpmeventlog implements event log parsing and replay for the PC Client
-// TPM PCR_based event log.
-// It supports both the SHA-1 only and crypto agile log formats.
 package tpmeventlog
 
 import (
+	"crypto"
 	"errors"
 
 	"github.com/google/go-eventlog/common"
 	"github.com/google/go-eventlog/extract"
 	pb "github.com/google/go-eventlog/proto/state"
-	"github.com/google/go-eventlog/register"
 	"github.com/google/go-eventlog/tcg"
+	"github.com/google/go-tpm/legacy/tpm2"
 )
 
 // ExtractOpts gives options for extracting information from an event log.
 type ExtractOpts struct {
 	Loader common.Bootloader
 }
 
-// ExtractFirmwareLogState parses a PC Client event log and replays the parsed
-// event log against the PCR bank specified by hash.
-//
-// It returns the corresponding FirmwareLogState containing the events verified
-// by particular PCR indexes/digests.
-// It returns an error on failing to replay the events against the PCR bank or
-// on failing to parse malformed events.
+// ExtractFirmwareLogState extracts event info from a verified TPM PC Client
+// event log into a FirmwareLogState.
+// It returns an error on failing to parse malformed events.
 //
 // The returned FirmwareLogState may be a partial FirmwareLogState.
 // In the case of a partially filled state, err will be non-nil.
 // Callers can look for individual errors using `errors.Is`.
 //
-// It is the caller's responsibility to ensure that the passed PCR values can be
-// trusted. Users can establish trust in PCR values by either calling
-// client.ReadPCRs() themselves or by verifying the values via a PCR quote.
-func ExtractFirmwareLogState(rawEventLog []byte, pcrBank register.PCRBank, opts ExtractOpts) (*pb.FirmwareLogState, error) {
+// It is the caller's responsibility to ensure that the passed events have
+// been replayed (e.g., using `tcg.ParseAndReplay`) against a verified measurement
+// register bank.
+func ExtractFirmwareLogState(events []tcg.Event, hash crypto.Hash, opts ExtractOpts) (*pb.FirmwareLogState, error) {
```

Comment:
Feels weird about having the same function signature in two packages (ccel and tpmeventlog).

Most of of two functions are the same, can we put this function in a separate package with a parameter passing into it (ccel/tpmeventlog)